### PR TITLE
Miscellaneous features and fixes for Nvidia GPU support

### PR DESCRIPTION
**Pull request by francescolavra**
This PR contains a set of new features and fixes that are needed in order to run the out-of-tree Nvidia GPU klib (https://github.com/nanovms/gpu-nvidia):
- stripping of debug information from klib binaries
- support for associating a major and minor number to Unix special files
- fix to prevent VGA console driver from attaching to non-VGA PCI display devices
- fix to context pause/resume functions to prevent assertion failures when kernel threads are run during kernel shutdown
- support for file-descriptor-specific mmap() implementations
- support for SOCK_SEQPACKET Unix domain sockets
- addition of table_remove() function
- fix to handling of empty file paths in syscalls such as unlink(2)

### CODE REVIEW

**Review by sanderssj — COMMENTED** [review]
LGTM! Nice work!

File: src/kernel/page.c [hunk 1]
```diff
@@ -360,7 +360,7 @@ void unmap_pages_with_handler(u64 virtual, u64 length, range_handler rh)
 
 #define next_addr(a, mask) (a = (a + (mask) + 1) & ~(mask))
 #define INDEX_MASK (PAGEMASK >> 3)
-static boolean map_level(u64 *table_ptr, int level, range v, u64 *p, u64 flags)
+static boolean map_level(u64 *table_ptr, int level, range v, u64 *p, u64 flags, flush_entry fe)
```
[the remark this answers is not in the document]
Comment (francescolavra) [reply src/kernel/page.c:366]: Sure, I added a comment to clarify the use of the flush_entry parameter

**Review by wjhun — APPROVED** [review]
Looks great!

File: src/runtime/list.h [hunk 2]
```diff
@@ -66,6 +72,14 @@ static inline void list_insert_before(struct list * pos,
     pos->prev = new;
 }
 
+static inline void list_replace(struct list *old, struct list *new)
```
[the remark this answers is not in the document]
Comment (francescolavra) [reply src/runtime/list.h:75]: OK, I changed list_move() so that it calls list_replace()

**Comment by francescolavra** [comment]
Oddly enough, the change (https://github.com/nanovms/nanos/pull/1783/commits/f57d4db156658067d38b99ce5c27b933bd93edd6) to filesystem_resolve_cstring() to return FS_STATUS_NOENT on empty an path has a side effect that makes a glibc bug (https://bugs.launchpad.net/ubuntu/+source/glibc/+bug/1982326) more invasive when running the thread runtime test program (even though filesystem_resolve_cstring() is never called with an empty path argument when running this program), which causes a test failure in Jenkins. As a workaround, I increased the sleep time in the thread test program.
